src/Esfa.Vacancy.Register.Api/App_Start/CustomExceptionHandler.cs

```diff
@@ -50,6 +52,17 @@ public override void Handle(ExceptionHandlerContext context)
                 return;
             }
 
+            if (context.Exception is InfrastructureException)
+            {
+                var response = new HttpResponseMessage(HttpStatusCode.InternalServerError);
+                response.Content = new StringContent(GenericErrorMessage);
+                context.Result = new CustomErrorResult(context.Request, response);
+
+                Logger.Error(context.Exception.InnerException, "Unexpected infrastructure error");
+
+                return;
+            }
+
```

Comment:
I know none of this is tested yet, but it looks testable by checking whats on the `context` that was passed in. What do you guys think about tests of app_start stuff?